src/celengine/observer.cpp

```diff
@@ -816,16 +814,15 @@ void Observer::setFrame(ObserverFrame::CoordinateSystem cs, const Selection& ref
 /*! Set the observer's reference frame. The position of the observer in
  *  universal coordinates will not change.
  */
-void Observer::setFrame(const ObserverFrame& f)
+void Observer::setFrame(const shared_ptr<const ObserverFrame>& f)
 {
-    if (frame != &f)
+    if (frame != f)
     {
-        auto* newFrame = new ObserverFrame(f);
+        auto newFrame = make_shared<ObserverFrame>(*f);
```

Comment:
`auto* newFrame = new ObserverFrame(f);` was required due to ownership and `new`/`delete` issues, now with `shared_ptr` we don't need this and instead of `newFrame` `f` should be used.